background/services/chain/serial-fallback-provider.ts

```
@@ -255,9 +248,78 @@ export default class SerialFallbackProvider extends JsonRpcProvider {
     return this.currentProvider.detectNetwork()
   }
 
-  // Overriding internal functionality here to support event listener
-  // restoration on reconnect.
-  // eslint-disable-next-line no-underscore-dangle
+  /**
+   * Subscribe to pending transactions that have been resolved to a full
+   * transaction object; uses optimized paths when the provider supports it,
+   * otherwise subscribes to pending transaction hashes and manually resolves
+   * them with a transaction lookup.
+   */
+  async subscribeFullPendingTransactions(
+    { address, network }: AddressOnNetwork,
+    handler: (pendingTransaction: AnyEVMTransaction) => void
+  ): Promise<void> {
+    if (this.evmNetwork.chainID !== network.chainID) {
+      logger.error(
+        `Tried to subscribe to pending transactions for chain id ` +
+          `${network.chainID} but provider was on ` +
+          `${this.evmNetwork.chainID}`
+      )
+      return
+    }
+
+    try {
+      await this.subscribe(
+        "filteredNewFullPendingTransactionsSubscriptionID",
+        ["alchemy_filteredNewFullPendingTransactions", { address }],
+        async (result: unknown) => {
+          // TODO use proper provider string
+          // handle incoming transactions for an account
+          try {
+            const transaction = transactionFromAlchemyWebsocketTransaction(
+              result,
+              network
+            )
+
+            handler(transaction)
+          } catch (error) {
+            logger.error(
+              `Error handling incoming pending transaction: ${result}`,
+              error
+            )
+          }
+        }
+      )
+    } catch (error) {
+      const errorString = String(error)
+      if (errorString.match(/unsupported subscription/i)) {
```

We are silently ignoring an error if the condition is false.

Perhaps
```typescript
      if (!errorString.match(/unsupported subscription/i)) throw error
```
or at the bare minimum we need a comment saying that we are ignoring errors purposfully.

---

Yikes, yep. Thanks for flagging.